src/main/java/uk/gov/hmcts/ccd/domain/model/search/elasticsearch/ElasticsearchRequest.java

```diff
@@ -59,22 +89,63 @@ public JsonNode getSource() {
                 .map(JsonNode::asText)
                 .map(this::getFieldId)
                 .filter(Objects::nonNull)
-                .collect(Collectors.toList());
+                .collect(toList());
         }
 
         return Collections.emptyList();
     }
 
     public boolean hasQuery() {
-        return searchRequest.has(QUERY);
+        return nativeSearchRequest.has(QUERY);
     }
 
     public JsonNode getQuery() {
-        return searchRequest.get(QUERY);
+        return nativeSearchRequest.get(QUERY);
+    }
+
+    public boolean hasRequestedSupplementaryData() {
+        return requestedSupplementaryData != null;
+    }
+
+    public void setRequestedSupplementaryData(ArrayNode requestedSupplementaryData) {
+        this.requestedSupplementaryData = requestedSupplementaryData;
     }
 
-    public String toJson() {
-        return searchRequest.toString();
+    public void setRequestedSupplementaryData(String... requestedSupplementaryDataFields) {
+        requestedSupplementaryData = MAPPER.createArrayNode();
+        for (String field : requestedSupplementaryDataFields) {
+            requestedSupplementaryData.add(new TextNode(field));
+        }
+    }
+
+    /**
+     * Creates a JSON string representing the Elasticsearch request object.
+     * Custom properties supported by CCD will be merged appropriately to generate a native Elasticsearch request.
+     * @return JSON string representing the Elasticsearch request object.
+     */
+    public String toFinalRequest() {
+        JsonNode mergedRequest = nativeSearchRequest.deepCopy();
+
+        ((ObjectNode) mergedRequest).set(SOURCE, mergedSourceFields());
```

Comment:
Contrary to before, any request now will have the _source field right?
Whereas before on when:
`elasticsearchRequest.hasSourceFields()`